### PR TITLE
Switch to HTTPSClientSession constructor available in Poco 1.14

### PR DESCRIPTION
**Pull request by jclarkeSTFC**
Following `Poco` updating to 1.14, the constructor we were using for `HTTPSClientSession` is no longer available, so I've changed it over to one that works, it's also neater.

I removed the pin for `Poco` 1.12.1, because we're now on 1.14, so no need for the old pin.

I also removed some `CMake` code that was causing compilation errors on Windows.

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

**Comment by jclarkeSTFC** [comment]
> The pin for Poco looks a little long. Is this an opportunity to simplify it?

This will fix the build failures instead of pinning Poco

**Comment by sf1919** [comment]
There was already a pin in for Poco before the current issues.

**Comment by sf1919** [comment]
This will need testing on error reporter, catalog etc.

**Review by cailafinn — DISMISSED** [review]
Tested catalog and reports sessions and all working as expected.

File: conda/recipes/conda_build_config.yaml [hunk 1]
```diff
@@ -85,7 +85,7 @@ orsopy:
 # 1.12.2 introduced an unguarded #define NOMINMAX which causes a compiler warning.
 # It's resolved on their devel branch but not yet included in a release, as of 1.12.5.
 poco:
-  - '1.12.1|>=1.12.6,!=1.14.0'
+  - '>=1.12.6'
```
[the remark this answers is not in the document]
Comment (jclarkeSTFC) [reply conda/recipes/conda_build_config.yaml]: Thanks for spotting that, I've made the changes

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**